torch/testing/_internal/common_utils.py

```diff
@@ -1666,6 +1666,8 @@ def _should_stop_test_suite(self):
             try:
                 torch.cuda.synchronize()
             except RuntimeError as rte:
+                from warnings import warn
+                warn(f"TEST SUITE EARLY TERMINATION due to torch.cuda.synchronize() failed with {rte}")
```

Comment:
Should this be surfaced as an error and not warning?

Reply:
Is there a way to make it more verbose?
For example, see:
```
% cat expectedFailure.py 
#!/usr/bin/env python3

from unittest import TestCase, expectedFailure, main

class ExpectedFailureTestCase(TestCase):
    @expectedFailure
    def test_fail(self):
        from warnings import warn
        warn("Foo")
        print("Bar")
        self.assertEqual(1, 0, "broken")

if __name__ == "__main__":
    main()
```
outputs something like:
```
expectedFailure.py:9: UserWarning: Foo
  warn("Foo")
Bar
x
----------------------------------------------------------------------
Ran 1 test in 0.000s
```
I.e. warning looks more prominent that exception

Reply:
Ah, I just meant more like, does this PR actually stop the test suite with an error or will it attempt to go on with a warning? 

I would suggest stopping the test suite with an error because if the context is wrong, the future tests will fail as well right?

Reply:
The test suite will be stopped, the warning here is to make logs indicate that something is wrong (as opposed to the situation that existed on master, where the test would be xfailed, all subsequent tests would be skipped, but logs would look absolutely normal, and unless you know that many more tests are expected to run, you wouldn't know that there's actually a failure preventing all subsequent tests from running). 
Even if the test in question is not xfailed, this PR makes the log more informative, saying "last test failed, and remaining tests weren't run", as opposed to "a single test failed".